File: 1/2020.09.22/144.Binary Tree Preorder Traversal/python/solution.py
class Solution:
    def preorderTraversal(self, root: TreeNode) -> List[int]:
        result = []
        self.traversal(root, result)
        return result

    def traversal(self, root, result):
        if root:
            result.append(root.val)
            self.traversal(root.left, result)
            self.traversal(root.right, result)

File: 1/2020.09.22/144.Binary Tree Preorder Traversal/java/solution.java
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        traversal(root, res);
        return res;
    }
    
    private void traversal(TreeNode root, List<Integer> res) {
        if (root != null) {
            res.add(root.val);
            traversal(root.left, res);
            traversal(root.right, res);
        }
    }
}
